src/Serials.h

```diff
@@ -17,16 +17,16 @@ namespace Serials {
   // just to satisfy linker in gcc I needed to add empty parentheses to other virtual methods...
   class AbstractSerial {
   public:
-    virtual void begin(int baudRate) = 0;
+    virtual void begin(int baudRate, uint32_t config, int pinRx, int pinTx) = 0;
     virtual Stream *getStream() = 0;
     virtual ~AbstractSerial() {};
   };
 
   struct Hardware: public AbstractSerial {
     Hardware(HardwareSerial &serial): serial(serial) {}
 
-    void begin(int baudRate) {
-      serial.begin(baudRate);
+    void begin(int baudRate = 9600, uint32_t config = SERIAL_8N1, int pinRx = -1, int pinTx = -1) {
+      serial.begin(baudRate, config, pinRx, pinTx);
```

Comment:
esp32 `HardwareSerial` is different - and if you try to compile for arduino this would not work
the solution is to put you modified API (with additional params) behind `#ifnded`